### PR TITLE
Sink keys doc

### PR DESCRIPTION
**Pull request by umanwizard**
Do not land until after #4668 goes in.

---
This change is [Reviewable](https://reviewable.io/reviews/materializeinc/materialize/4680)

### CODE REVIEW

File: doc/user/content/sql/create-sink.md [hunk 1]
```diff
@@ -31,14 +31,15 @@ _item&lowbar;name_ | The name of the source or view you want to send to the sink
...
-{{< diagram "kafka-connector.svg" >}}
+{{< diagram "sink-kafka-connector.svg" >}}
 
 Field | Use
 ------|-----
 **KAFKA BROKER** _host_ | The Kafka broker's host name.
 **TOPIC** _topic&lowbar;prefix_ | The prefix used to generate the Kafka topic name to create and write to.
 **WITH OPTIONS (** _option&lowbar;_ **)** | Options affecting sink creation. For more details see [`WITH` options](#with-options).
 **CONFLUENT SCHEMA REGISTRY** _url_ | The URL of the Confluent schema registry to get schema information from.
+**KEYS (** _key&lowbar;column&lowbar;list_ **)** | An optional list of columns to use for the Kafka key.
```
Comment (sploiselle) [inline doc/user/content/sql/create-sink.md:42]: This is likely my own ignorance w/r/t Kafka, but would it be useful to describe what this does in Materialize in the details section of this page?

File: doc/user/content/release-notes.md [hunk 2]
```diff
@@ -53,6 +53,8 @@ Wrap your release notes at the 80 character mark.
 
 - Add the `regexp_match` function to search a string with a regular expression.
 
+- Allow adding keys to [sinks](/sql/create-sink).
```
Comment (sploiselle) [inline doc/user/content/release-notes.md:56]: include that they're only applicable to Kafka sinks

File: doc/user/content/sql/create-sink.md (hunk 1 above)
Comment (umanwizard) [reply doc/user/content/sql/create-sink.md:42]: It doesn't do anything in Materialize per se - only in Kafka.

In Kafka every record has a key and a value; before this change, the key would always be "null" in sinks that we produce. Now the key is composed of whatever values from the record that the user specifies in the `KEYS` clause (which we are renaming to `KEY`, btw - I will update this PR accordingly).

Do you know of any better way to express it?

Comment (sploiselle) [reply doc/user/content/sql/create-sink.md:42]: Ahh, that makes sense. I would assume folks actually running Kafka to understand that concept, so strike my comment.

**Pull request merged**